src/actions/sources/newSources.js

```diff
@@ -201,6 +201,11 @@ export function newSource(source: Source) {
 export function newSources(sources: Source[]) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     sources = sources.filter(source => !getSource(getState(), source.id));
+    if (!prefs.chromeAndExtenstionsEnabled) {
+      sources = sources.filter(
+        source => !(source.url && source.url.startsWith("moz-extension://"))
+      );
+    }
```

Comment:
i think we could move this flag to a selector like `getRelativeSources`, which will just decide what is shown in the tree and modal.